src/cr/cube/smoothing.py

```diff
@@ -7,12 +7,35 @@
 from cr.cube.enums import DIMENSION_TYPE as DT
 
 
-class SingleSidedMovingAvgSmoother(object):
+class BaseSmoothingSpec(object):
+    """Base object class for Smoother variants."""
+
+    def __init__(self, dimension):
+        self._dimension = dimension
+
+    @classmethod
+    def smoother(cls, dimension):
+        """Returns appropriate Smoother object according to passed function.
+
+        Raises an exception if the function is different from `one_sided_moving_avg`
+        that at the moment is the only one implemented.
+        """
+        smoothing_dict = dimension.smoothing_dict
+        function = smoothing_dict.get("function") or "one_sided_moving_avg"
+        if function != "one_sided_moving_avg":
+            raise NotImplementedError("Function {} is not available.".format(function))
+        return _SingleSidedMovingAvgSmoother(
+            window=smoothing_dict.get("window") or 2,
+            dimension_type=dimension.dimension_type,
+        )
```

Comment:
Returns directly the smoother and not the spec cause the smoother is the actual object we need to apply the algorithm to the second order measures in the measure.py module